File: 02Algorithm_Study/04_week_homeWork/spy.cpp
#include <string>
#include <vector>
#include <unordered_map>
using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;

    unordered_map <string, int> m;


    for( auto i : clothes){
        if(m.find(i[1]) == m.end()){
            m.insert(make_pair(i[1],1));
        }
        else{
            m.find(i[1]) ->second++;
        }
    }


    for(auto i = m.begin() ; i != m.end(); i++){
        answer = answer * (i->second + 1);

    }
    return answer - 1;
}

File: 02Algorithm_Study/04_week_homeWork/spy.py
import collections

def solution(clothes):
    answer = 1
    dic = dict()
    for i in range(len(clothes)):
        k = clothes[i][1]
        v = clothes[i][0]
        if( k in dic):
            dic[k].append(v)
        else:
            dic[k] = [v]

    
    for i in dic.keys():
        answer = answer * (len(dic[i]) + 1)
    return answer - 1


print(solution([["yellow_hat", "headgear"], ["blue_sunglasses", "eyewear"], ["green_turban", "headgear"]]))
